nr_logger.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import time
+
+from process import Process
+
+
+class LogWriter(Process):
+  """ Reads data from the logging queue and writes it to a file. """
+
+  def __init__(self, messenger, filename):
+    """ Args:
+      messenger: The Messanger instance to use.
+      filename: The base file name to write log messages to. """
+    super(LogWriter, self).__init__(messenger)
+
+    # Use the current system time to make each new file unique.
+    systime = int(time.time())
+    self.__file = open("%s-%d.log" % (filename, systime), "a")
+
+  def __del__(self):
+    self.__file.close()
+
+  def _run(self):
+    log_queue = self._messenger.get_queue("logging")
```

Comment:
So, when messenger is passed in and the process is forked, it's actually copied, so technically, the version of messenger that this process is accessing is local and is not tied to the messenger instance in the main process. As for actually grabbing the queue, queues are designed specifically to be passed around in this way, so that is not a problem.